docs/en/engines/table-engines/mergetree-family/mergetree.md

```diff
@@ -471,8 +471,8 @@ The `set` index can be used with all functions. Function subsets for other index
 | [notEmpty](/docs/en/sql-reference/functions/array-functions#function-notempty)                          | ✔           | ✔      | ✗           | ✗           | ✗             |
 | hasToken                                                                                                   | ✗           | ✗      | ✗           | ✔           | ✗             |
 | hasTokenOrNull                                                                                                   | ✗           | ✗      | ✗           | ✔           | ✗             |
-| hasTokenCaseInsensitive                                                                                                   | ✗           | ✗      | ✗           | ✔           | ✗             |
-| hasTokenCaseInsensitiveOrNull                                                                                                   | ✗           | ✗      | ✗           | ✔           | ✗             |
+| hasTokenCaseInsensitive (must be indexed as lowercase)                                                                                                   | ✗           | ✗      | ✗           | ✔           | ✗             |
```

Comment:
Does it actually index tokens in lowercase?

Reply:
Certainly not automagically, but the index of the test data generated in the functional test file I modified is as follows:
```
INDEX tok_bf (s, lower(s)) TYPE tokenbf_v1(512, 3, 0) GRANULARITY 1
```

I found that both `hasToken` and `hasTokenCaseInsensitive` work when a column is indexed this way and the changes I submitted are applied.

Reply:
I think the docs should give an example what "must be indexed as lowercase" means. Please
- remove `(must be indexed as lowercase)` from l. 474, as a replacement, simply add `(*)`, and 
- add a sentence `(*) For hasTokenCaseInsensitve and hasTokenCaseInsensitive to be effective, the data skipping index of type tokenbf_v1 must be created on lowercased data, for example CREATE TABLE tab (id UInt64, s String, INDEX tok_bf_idx (s, lower(s)) TYPE tokenbf_v1(512, 3, 0) GRANULARITY 1) ... .
) ENGINE = MergeTree() ` to l. 478.

Reply:
How do we check that it was created on lowercase data?

Reply:
Well.. is it possible to get ahold of the expression that was used to generate the index? Or perhaps we could flag the index upon creation that it supports lowercase lookup?

If one of these sounds acceptable then I should be able to sort something out.